app/src/webkit/java/org/mozilla/focus/web/WebViewProvider.java

```diff
@@ -67,6 +68,48 @@ private static void configureSettings(Context context, WebSettings settings) {
         settings.setAllowFileAccess(false);
 
         settings.setBlockNetworkImage(appSettings.shouldBlockImages());
+
+        settings.setUserAgentString(buildUserAgentString(context, settings));
+    }
+
+    private static String buildUserAgentString(final Context context, final WebSettings settings) {
+        final StringBuilder uaBuilder = new StringBuilder();
+
+        uaBuilder.append("Mozilla/5.0");
+
+        // WebView by default includes "; wv" as part of the platform string, but we're a full browser
+        // so we shouldn't include that.
+        // Most webview based browsers (and chrome), include the device name AND build ID, e.g.
+        // "Pixel XL Build/NOF26V", that seems unnecessary (and not great from a privacy perspective),
+        // so we skip that too.
+        uaBuilder.append(" (Linux; Android ").append(Build.VERSION.RELEASE).append(") ");
+
+        // Use the default WebView agent string here for everything after the platform, but insert
+        // Focus in front of Chrome.
+        // E.g. a default webview UA string might be:
+        // Mozilla/5.0 (Linux; Android 7.1.1; Pixel XL Build/NOF26V; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/56.0.2924.87 Mobile Safari/537.36
+        // And we reuse everything from AppleWebKit onwards, except for adding Focus.
+        final String existing = settings.getUserAgentString();
+        int start = existing.indexOf("AppleWebKit");
```

Comment:
Is it guaranteed that this string will always be present? Maybe we should parse for tokens like "Something/Version (Optional stuff)".